ios/Exponent/Kernel/ReactAppManager/EXReactAppManager.mm

```diff
@@ -421,12 +422,58 @@ - (void)_handleJavaScriptStartLoadingEvent:(NSNotification *)notification
   });
 }
 
+/** Returns a value like `http://localhost:19000 — my-app` */
+- (NSString *)getBestContextName
+{
+  try {
+    NSURL *bundleUrl = [self bundleUrl];
+    NSURLComponents *urlComponents = [[NSURLComponents alloc] initWithURL:bundleUrl resolvingAgainstBaseURL:NO];
+    // Strip the extra components.
+    urlComponents.path = nil;
+    urlComponents.query = nil;
+    
+    
+    NSString *hostnameUrl = [urlComponents string];
+    
+    // Strip scheme to emulate Safari, NSURLComponents doesn't support this by default.
+    NSRange dividerRange = [hostnameUrl rangeOfString:@"://"];
+    NSUInteger divide = NSMaxRange(dividerRange);
+    hostnameUrl = [hostnameUrl substringFromIndex:divide];
+  
+    // Attempt to use the manifest name, otherwise fallback on the base URL.
+    // This closely emulates the browser, which uses the document.title when possible.
+    if (_appRecord.appLoader.manifest != nil) {
+      NSString *name = [_appRecord.appLoader.manifest name];
+      if (name != nil) {
+        // Like `http://localhost:19000 — my-app`
+        hostnameUrl = [hostnameUrl stringByAppendingString:[@" — " stringByAppendingString:name]];
+      }
+    }
+    return hostnameUrl;
+  } catch (NSException *ex) {
+    RCTLogWarn(@"Failed to fetch page name: %@", ex.reason);
+    return @"Cannot Open Page";
+  }
```

Comment:
`@try-catch` is sparingly used in Objective-C. It usually indicates the code is broken, not that something with the system went wrong. Either remove this or guard just the specific line that needs guarding, though I suspect any method that can fail assuming correct code will take an NSError double pointer.